src/basic/namespace-util.c

```diff
@@ -479,6 +480,24 @@ int namespace_open_by_type(NamespaceType type) {
         return fd;
 }
 
+int namespace_is_init(NamespaceType type) {
+        assert(type >= 0);
+        assert(type <= _NAMESPACE_TYPE_MAX);
+
+        if (namespace_info[type].root_inode == 0)
+                return -EBADR; /* Cannot answer this question */
+
+        _cleanup_close_ int fd = namespace_open_by_type(type);
+        if (fd < 0)
+                return fd;
+
+        struct stat st;
+        if (fstat(fd, &st) < 0)
```

Comment:
Why not to call `stat()` with `pid_namespace_path()`, like we do in `is_our_namespace()` ?